src/extension.ts

```diff
@@ -103,3 +135,107 @@ export async function activateInternal(context: vscode.ExtensionContext, perfSta
 // tslint:disable-next-line:no-empty
 export function deactivateInternal(): void {
 }
+
+async function setupLocalProjectFolder(uri: vscode.Uri, filePath: string, token: string, account: AzureAccount): Promise<void> {
+    const requiredInputs = getAllRequiredInputs(uri.query);
+    const resourceId: string | null = requiredInputs.resourceId;
+    const defaultHostName: string | null = requiredInputs.defaultHostName;
+    const devContainerName: string | null = requiredInputs.devContainerName;
+    const language: string | null = requiredInputs.language;
+    const downloadAppContent: string | null = requiredInputs.downloadAppContent;
+
+    if (resourceId && defaultHostName && devContainerName && language && downloadAppContent) {
+        ext.context.globalState.update('projectLanguage', getProjectLanguageForLanguage(language));
+        const vsCodeFilePathUri: vscode.Uri = vscode.Uri.file(filePath);
+        const toBeDeletedFolderPathUri: vscode.Uri = vscode.Uri.joinPath(vsCodeFilePathUri, 'tobedeleted');
+
+        try {
+            const functionAppName: string = getNameFromId(resourceId);
+            const downloadFilePath: string = vscode.Uri.joinPath(toBeDeletedFolderPathUri, `${functionAppName}.zip`).fsPath;
```

Comment:
should we handle the scenario where the selected folder already contains a functionapp and/or the downloaded zip?

Reply:
Not sure I understand the comment here.

Reply:
As discussed offline, it just overwrites the old one.